Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrTextEmitter.kt

```diff
@@ -0,0 +1,669 @@
+                    it
+                }else{
+                    "`$it`"
+                }
+            }
+            out("package $packageName")
+            out("")
+        }
+        if (context.platform == KotlinPlatform.NATIVE) {
+            out("import kotlin.native.SymbolName")
+            out("import kotlinx.cinterop.internal.*")
+        }
+        out("import kotlinx.cinterop.*")
+
+        kotlinFile.buildImports().forEach {
+            out(it)
+        }
+
+        out("")
+
+        out("// NOTE THIS FILE IS AUTO-GENERATED")
+    }
+    fun emit(ktFile: Appendable, cFile: Appendable, entryPoint: String?) {
+
+        withOutput(cFile) {
+            context.libraryForCStubs.preambleLines.forEach {
+                out(it)
+            }
+            out("")
+
+            out("// NOTE THIS FILE IS AUTO-GENERATED")
+            out("")
+
+            nativeBridges.nativeLines.forEach(out)
+
+            if (entryPoint != null) {
+                out("extern int Konan_main(int argc, char** argv);")
+                out("")
+                out("__attribute__((__used__))")
+                out("int $entryPoint(int argc, char** argv)  {")
+                out("  return Konan_main(argc, argv);")
+                out("}")
+            }
+        }
+
+        // Stubs generation may affect imports list so do it before header generation.
+        val stubLines = generateKotlinFragmentBy {
+            printer.visitSimpleStubContainer(builderResult.stubs, null)
+        }
+
+        withOutput(ktFile) {
+            emitKotlinFileHeader()
+            stubLines.forEach(out)
+            nativeBridges.kotlinLines.forEach(out)
+            if (context.platform == KotlinPlatform.JVM) {
+                out("private val loadLibrary = System.loadLibrary(\"${context.libName}\")")
+            }
+        }
+    }
+    private val printer = object : StubIrVisitorAux<StubContainer?, Unit> {
+
+        override fun visitClass(element: ClassStub, owner: StubContainer?) {
+            element.annotations.forEach {
+                out(renderAnnotation(it))
+            }
+            val header = renderClassHeader(element)
+            when {
+                element is ClassStub.Simple && element.children.isEmpty() -> out(header)
+                element is ClassStub.Companion && element.children.isEmpty() -> out(header)
+                else -> block(header) {
+                    if (element is ClassStub.Enum) {
+                        emitEnumBody(element)
+                    } else {
+                        element.children.forEach {
+                            out("")
+                            it.accept(this, element)
+                        }
+                    }
+                }
+            }
+        }
+
+        override fun visitTypealias(element: TypealiasStub, owner: StubContainer?) {
+            out("typealias ${renderClassifierDeclaration(element.alias.classifier)} = ${renderStubType(element.aliasee)}")
```

Comment:
> `renderClassifierDeclaration(element.alias.classifier)`

Somewhat inconsistent with `element.alias` having type arguments.
Either make `element.alias` classifier or render the full type properly here.